### ISSUE TITLE
merging deep learning models

### ISSUE THREAD

**Issue by Yogeskumaran** [keras-team/keras#19147]
i have been working on a deep learning project for ai vs human text classification. my machine is not enough to process lakhs of data so i divided the data into chunks and created a individual model for each chunks. Totally five models where created and now i want to merge all those 5 models into a one single model and i merged that also . but the issue is because of merging the 5 models the new merged model expects 5 input but i want the new model to be expecting single input . how to achieve this?
![Uploading Screenshot 2024-02-05 120025.png…]()

**Comment by sachinprasadhs** [comment]
Screenshot which you have mentioned seems to be broken, could you please provide more details about the issue. Thanks!

**Comment by Yogeskumaran** [comment]
> Screenshot which you have mentioned seems to be broken, could you please provide more details about the issue. Thanks!

from keras.models import load_model
from keras.models import Sequential, Model
from keras.layers import Dense, LSTM, Embedding, concatenate, Input
from tensorflow.keras.layers import Input, Embedding, Conv1D, GlobalMaxPooling1D, Dense, Dropout, LSTM
model3=load_model('dmodel3.keras')
model6=load_model('dmodel6.keras')
model7=load_model('dmodel7.keras')
model8=load_model('dmodel8.keras')
model9=load_model('dmodel9.keras')
input3=Input(shape=(1496,))
input6=Input(shape=(1553,))
input7=Input(shape=(1706,))
input8=Input(shape=(1693,))
input9=Input(shape=(1718,))
embedding3 = Embedding(input_dim=51000, output_dim=100)(input3)
embedding6 = Embedding(input_dim=51000, output_dim=100)(input6)
embedding7 = Embedding(input_dim=51000, output_dim=100)(input7)
embedding8 = Embedding(input_dim=51000, output_dim=100)(input8)
embedding9 = Embedding(input_dim=51000, output_dim=100)(input9)
conv1d3 = Conv1D(128, 5, activation='relu')(embedding3)
maxpool3 = GlobalMaxPooling1D()(conv1d3)
dense3 = Dense(64, activation='relu')(maxpool3)
dropout3 = Dropout(0.5)(dense3)

conv1d6 = Conv1D(128, 5, activation='relu')(embedding6)
maxpool6 = GlobalMaxPooling1D()(conv1d6)
dense6 = Dense(64, activation='relu')(maxpool6)
dropout6 = Dropout(0.5)(dense6)
3
conv1d7 = Conv1D(128, 5, activation='relu')(embedding7)
maxpool7 = GlobalMaxPooling1D()(conv1d7)
dense7 = Dense(64, activation='relu')(maxpool7)
dropout7 = Dropout(0.5)(dense7)

conv1d8 = Conv1D(128, 5, activation='relu')(embedding8)
maxpool8 = GlobalMaxPooling1D()(conv1d8)
dense8 = Dense(64, activation='relu')(maxpool8)
dropout8 = Dropout(0.5)(dense8)

conv1d9 = Conv1D(128, 5, activation='relu')(embedding9)
maxpool9 = GlobalMaxPooling1D()(conv1d9)
dense9 = Dense(64, activation='relu')(maxpool9)
dropout9 = Dropout(0.5)(dense9)
merged=concatenate([dropout3,dense3,dropout6,dense6,dropout7,dense7,dropout8,dense8,dropout9,dense9])
shared_dense = Dense(32, activation='relu')(merged)
output3=Dense(1,activation='sigmoid')(shared_dense)
output6=Dense(1,activation='sigmoid')(shared_dense)
output7=Dense(1,activation='sigmoid')(shared_dense)
output8=Dense(1,activation='sigmoid')(shared_dense)
output9=Dense(1,activation='sigmoid')(shared_dense)
merged_model3=Model(inputs=[input3,input6,input7,input8,input9],outputs=[output3,output6,output7,output8,output9])
this is my code creating several models and want to merge that into one single model,but if i done that i merged 5 models so final model asks me 5 inputs ,i want the efficiency of that 5 models into final model and the input and output should be only ony. how do i do that ,what alterations should i alter that in my code

**Comment by sachinprasadhs** [comment]
You can concatenate all the outputs and then feed it to your model, which is something like below.

```python
outputs = concatenate([output3,output6,output7,output8,output9])
merged_model3=Model(inputs=[input3,input6,input7,input8,input9],outputs=outputs)
```

**Comment by Yogeskumaran** [comment]
> You can concatenate all the outputs and then feed it to your model, which is something like below.
> 
> ```python
> outputs = concatenate([output3,output6,output7,output8,output9])
> merged_model3=Model(inputs=[input3,input6,input7,input8,input9],outputs=outputs)
> ```

but it asks 5 input??

**Comment by nimonkaranurag** [comment]
`input_combined = Input(shape=(max_input_length,))
shared_embedding = Embedding(input_dim=51000, output_dim=100)(input_combined)
conv1d_combined = Conv1D(128, 5, activation='relu')(shared_embedding)
maxpool_combined = GlobalMaxPooling1D()(conv1d_combined)
dense_combined = Dense(64, activation='relu')(maxpool_combined)
dropout_combined = Dropout(0.5)(dense_combined)
dropout3 = Dropout(0.5)(dense_combined)
dropout6 = Dropout(0.5)(dense_combined)
dropout7 = Dropout(0.5)(dense_combined)
dropout8 = Dropout(0.5)(dense_combined)
dropout9 = Dropout(0.5)(dense_combined)
merged = concatenate([dropout3, dropout6, dropout7, dropout8, dropout9])
shared_dense = Dense(32, activation='relu')(merged)
output3 = Dense(1, activation='sigmoid')(shared_dense)
output6 = Dense(1, activation='sigmoid')(shared_dense)
output7 = Dense(1, activation='sigmoid')(shared_dense)
output8 = Dense(1, activation='sigmoid')(shared_dense)
output9 = Dense(1, activation='sigmoid')(shared_dense)
merged_model = Model(inputs=[input_combined], outputs=[output3, output6, output7, output8, output9])
`
